src/utils.ts

```diff
@@ -38,8 +38,14 @@ export function findSMDataSources(): Array<DataSourceInstanceSettings<SMOptions>
   }) as unknown as Array<DataSourceInstanceSettings<SMOptions>>;
 }
 
-export function findLinkedDatasource(uid: string): DataSourceInstanceSettings | undefined {
-  return Object.values(config.datasources).find((ds) => ds.uid === uid);
+export function findLinkedDatasource(linkedDSInfo: LinkedDatasourceInfo): DataSourceInstanceSettings | undefined {
+  if (linkedDSInfo.uid) {
+    const linkedDS = Object.values(config.datasources).find((ds) => ds.uid === linkedDSInfo.uid);
+    if (linkedDS) {
+      return linkedDS;
+    }
+  }
+  return config.datasources[linkedDSInfo.grafanaName];
```

Comment:
I know this is a fallback, but did you mention it was not reliable to find the datasource by the `grafanaName`? should we add a comment to this line indicating this?

Reply:
Good idea! I've added a comment in above the datasource methods and linked to the description I've now filled out in this PR.